tests/ui/type/clarify-error-for-generics-with-default-issue-120785.rs

```diff
@@ -0,0 +1,11 @@
+struct What<W = usize, X = Vec<W>>(W, X);
+
+fn main() {
+    let mut b: What<usize> = What(5, vec![1, 2, 3]);
+    let c: What<usize, String> = What(1, String::from("meow"));
+    b = c; //~ ERROR mismatched types
+
+    let mut e: What<usize> = What(5, vec![1, 2, 3]);
+    let f: What<usize, Vec<String>> = What(1, vec![String::from("meow")]);
```

Comment:
Would it make sense to change the order in this second case, so the non-default-after-default case is first?

Reply:
I'm sorry, I'm not sure if I understand this correctly. Are you asking about swapping `usize` and `Vec<String>`?

Reply:
Sorry, no, the item order, e.g.:
```rust 
    let e: What<usize, Vec<String>> = What(1, vec![String::from("meow")]);
    let mut f: What<usize> = What(5, vec![1, 2, 3]);
    e = f;
```
That way you have a case (`b = c`) with `generics1.check_concrete_type_after_default(self.tcx, sub1)` is true and a case (`e = f`) with `generics2.check_concrete_type_after_default(self.tcx, sub2)` is true. It's just a minor improvement in code coverage, I think.

Reply:
Updated. Thanks for the clarification.